### ISSUE TITLE
Add EXPECTED_NODE_COUNT config item

### ISSUE THREAD

**Issue by ethomas997** [RotorHazard/RotorHazard#532]
If number of nodes found does not equal EXPECTED_NODE_COUNT then show warning message at top of 'Settings' page.  If EXPECTED_NODE_COUNT is zero (default value) then don't show any warning.

**Comment by HazardCreative** [comment]
What sets this value? Saved on previous startup? Perhaps a config option, with a value to prevent the warning for development (maybe false or null)? I'd lean toward keeping this out of the front-end UI.

**Comment by ethomas997** [comment]
I see it as being in the 'config.json' file, configured by the operator to the expected # of nodes.

**Comment by pulquero** [comment]
What is the problem this is trying to solve? Here's a problem I've had. I2C bus has a failure, RH starts. I fiddle around in the UI trying to wonder why it is behaving strangely, then I realise that RH has switched to mock node mode. So would we be better served if this was actually EXPECTED_I2C_NODE_COUNT? Does serial suffer the same affliction to the same extent?

**Comment by HazardCreative** [comment]
Currently if a serial node isn't found, RH refuses to start up entirely.

I'd actually like to see that behavior changed so I can develop with or without my desktop node attached and not have to change the config each time. (It's especially annoying since JSON doesn't provide for comments.)

**Comment by ethomas997** [comment]
It aborts the server if a configured serial node is not found to make it clear to the operator that something is wrong.  It's already all too common that we have operators asking for support while unaware that the server is actually using mock nodes.  (The latest code shows a warning message at the top of the 'Settings' page to try and help with that.)

I think we've talked about adding a parameter option to specify the config filename.  Another way would be to have batch files in your setup that copy in a specific config file before launching the server.

**Comment by pulquero** [comment]
Would it not make sense to never switch to mock nodes unless explicitly configured? That would seem to alleviate the root issue. Atm we seem to be dancing around it with ui elements and config.

**Comment by ethomas997** [comment]
Yeah, adding a config param flag for "use mock nodes" sounds good to me.

BTW, if the S32_PCB is detected (via GPIO25 pull-down) and no nodes are found then the server is still started up and run, so that the node processor can be updated via the "Update Nodes" button on the web GUI.  (A message is shown at the top of the Settings page in that case.)
